app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -55,24 +55,42 @@ def validate_authentication_status
       {:available => true, :message => nil}
     end
 
+    def self.verify_ssl_mode(security_protocol)
+      case security_protocol
+      when nil, '', 'non-ssl', 'ssl-without-validation'
+        OpenSSL::SSL::VERIFY_NONE
+      else
+        OpenSSL::SSL::VERIFY_PEER
+      end
+    end
+
+    def self.entrypoint(host, port, security_protocol)
+      return URI::HTTP.build(:host => host, :port => port.to_i).to_s if security_protocol == 'non-ssl'
+      URI::HTTPS.build(:host => host, :port => port.to_i).to_s
+    end
+
     # Hawkular Client
-    def self.raw_connect(hostname, port, username, password)
-      entrypoint = URI::HTTP.build(:host => hostname, :port => port.to_i).to_s
+    def self.raw_connect(host, port, username, password, security_protocol, cert_store)
       credentials = {
         :username => username,
         :password => password
       }
       options = {
-        :tenant => 'hawkular'
+        :tenant         => 'hawkular',
+        :verify_ssl     => verify_ssl_mode(security_protocol),
+        :ssl_cert_store => cert_store
       }
-      ::Hawkular::Client.new(:entrypoint => entrypoint, :credentials => credentials, :options => options)
+      ::Hawkular::Client.new(:entrypoint => entrypoint(host, port, security_protocol),
+                             :credentials => credentials, :options => options)
     end
 
     def connect(_options = {})
       @client ||= self.class.raw_connect(hostname,
                                          port,
                                          authentication_userid('default'),
-                                         authentication_password('default'))
+                                         authentication_password('default'),
+                                         security_protocol,
```

Comment:
I didn't know about that. I was actually wondering about it.
I'll update the code. Thanks.